libbeat/autodiscover/builder/helper.go

```diff
@@ -86,7 +86,20 @@ func GetHintAsList(hints common.MapStr, key, config string) []string {
 
 // GetProcessors gets processor definitions from the hints and returns a list of configs as a MapStr
 func GetProcessors(hints common.MapStr, key string) []common.MapStr {
-	return GetConfigs(hints, key, "processors")
+	processors := GetConfigs(hints, key, "processors")
+	for _, proc := range processors {
+		for key, value := range proc {
+			if str, ok := value.(string); ok {
+				cfg := common.MapStr{}
+				if err := json.Unmarshal([]byte(str), &cfg); err != nil {
+					logp.Debug("autodiscover.builder", "unable to unmarshal json due to error: %v", err)
```

Comment:
Does this need to be logp.Debugf?

Reply:
Checked the code, seems `logp.Debug` is correct. It accepts parameters for the format string.